src/Workspaces/Core/Portable/Utilities/TaskExtensions.cs

```diff
@@ -264,6 +264,22 @@ TResult outerFunction(Task t)
                 cancellationToken, taskContinuationOptions, scheduler).Unwrap();
         }
 
+        public static Task<TNResult> ContinueWithAfterDelay<TNResult>(
+            this Task task,
+            Func<TNResult> continuationFunction,
+            CancellationToken cancellationToken,
+            int millisecondsDelay,
+            TaskContinuationOptions taskContinuationOptions,
+            TaskScheduler scheduler)
+        {
+            Contract.ThrowIfNull(continuationFunction, nameof(continuationFunction));
+
+            return task.SafeContinueWith(t =>
+                Task.Delay(millisecondsDelay, cancellationToken).SafeContinueWith(
+                    _ => continuationFunction(), cancellationToken, TaskContinuationOptions.None, scheduler),
```

Comment:
📝 These calls were already allocating in the old implementation, but the new overloads allow callers to avoid further allocations.

Reply:
I don't understand this comment. What do you mean by "the new overloads allow callers to avoid further allocations"?
(still jetlagging, so my brain might not functioning properly at the moment :P)

Reply:
I created a new method to avoid allocations, but the method where I made this comment still has closure allocations. I was clarifying that I did not address allocations within this method, but instead I addressed allocations within the _caller_ of this method (i.e. a partial but measurable improvement over the original code).

Reply:
Got it. Thanks for the explanation.